packages/flutter_tools/lib/src/version.dart

```diff
@@ -130,17 +130,34 @@ class FlutterVersion {
   };
 
   /// A date String describing the last framework commit.
-  String get frameworkCommitDate => _latestGitCommitDate();
+  String get frameworkCommitDate => _latestGitCommitDate(lenient: true);
 
-  static String _latestGitCommitDate([ String branch ]) {
+  static String _latestGitCommitDate({
+    String branch,
+    bool lenient = false,
+  }) {
     final List<String> args = gitLog(<String>[
       if (branch != null) branch,
       '-n',
       '1',
       '--pretty=format:%ad',
       '--date=iso',
     ]);
-    return _runSync(args, lenient: false);
+    try {
+      // Don't plumb 'lenient' through directly so that we can print an error
+      // if something goes wrong.
+      return _runSync(args, lenient: false);
+    } on VersionCheckError catch (e) {
+      if (lenient) {
+        final DateTime dummyDate = DateTime.fromMillisecondsSinceEpoch(0);
+        printError('Failed to find the latest git commit date: $e\n'
```

Comment:
Should we prompt them to re-download the SDK?

Reply:
Also, does this print with `--machine`?  We should avoid printing anything that isn't parsable like https://github.com/flutter/flutter/issues/29209

Reply:
I think it should probably print unconditionally. An automatic process that parses the version output should probably fail if the info is just a placeholder.